h5py/tests/old/test_dataset.py

```diff
@@ -746,15 +746,15 @@ class TestStrings(BaseDataset):
 
     def test_vlen_bytes(self):
```

Comment:
What about some `check_string_dtype()` (and `check_vlen_dtype()`) here and in the following tests?

Reply:
I've just added some in `test_datatype` - does that cover what you meant?

Reply:
Not exactly (I thought about `check_*_dtype()` calls after `create_dataset()`, in these tests), but now it's definitely good enough to be merged.

Reply:
Ah, you want to check that they roundtrip correctly through HDF5. That makes sense. I've added some tests.